### ISSUE TITLE
JsonIgnoreProperties does not prevent Exception Conflicting getter/setter definitions for property

### ISSUE THREAD

**Issue by gmkll** [FasterXML/jackson-databind#426]
The use case is  a base class which I could not change and I want to ignore the conflicting property (putting JsonIgnoreProperties in a child class). Should the JsonIgnoreProperties not prevent this exception?? This occurs in this simplified test scenario:

The Mapping class:

``` java
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@JsonIgnoreProperties({ "userId"})
public class User {
 Integer userId; 

    void setUserId(String id) {
        setUserId(new Integer(id));
    };

    public Integer getUserId() 
    {
        return userId;
    }

    public void setUserId(Integer v)
    {
        this.userId = v;
    }
}
```

The Test:

``` java
    @Test
    public void testDeSerObject() throws Exception {
        ObjectMapper mapper = new ObjectMapper();
        JsonIgnoreProperties ignore = User.class.getAnnotation(JsonIgnoreProperties.class);
        System.out.println("ignore:,"+ ignore);
        String jsonString = "{id: 9, firstName: \"Mike\" }";
        User result = mapper.reader( User.class ).readValue(jsonString);
        System.out.println("result:"+ result.getUserId());
    }
```

This fails with Exception (cause) message:

``` java
Caused by: java.lang.IllegalArgumentException: Conflicting setter definitions for property "userId": org.apache.fulcrum.json.jackson.User#setUserId(1 params) vs org.apache.fulcrum.json.jackson.User#setUserId(1 params)
    at com.fasterxml.jackson.databind.introspect.POJOPropertyBuilder.getSetter(POJOPropertyBuilder.java:222)
    at com.fasterxml.jackson.databind.introspect.POJOPropertyBuilder.getMutator(POJOPropertyBuilder.java:295)
    at com.fasterxml.jackson.databind.introspect.BasicBeanDescription.findBackReferenceProperties(BasicBeanDescription.java:368)
    at com.fasterxml.jackson.databind.deser.BeanDeserializerFactory.addReferenceProperties(BeanDeserializerFactory.java:641)
    at com.fasterxml.jackson.databind.deser.BeanDeserializerFactory.buildBeanDeserializer(BeanDeserializerFactory.java:271)
    at com.fasterxml.jackson.databind.deser.BeanDeserializerFactory.createBeanDeserializer(BeanDeserializerFactory.java:168)
    at com.fasterxml.jackson.databind.deser.DeserializerCache._createDeserializer2(DeserializerCache.java:401)
    at com.fasterxml.jackson.databind.deser.DeserializerCache._createDeserializer(DeserializerCache.java:354)
    at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCache2(DeserializerCache.java:267)
```

**Comment by jharen** [comment]
I, too am hitting this issue with jackson-databind 2.6.3.  When I try to scan classes from Amazon's com.amazonaws.services.cloudformation.model package, some fail. An example is the com.amazonaws.services.cloudformation.model.Stack class, which defines:    

    private String stackStatus;
    ...
    public void setStackStatus(String stackStatus) {
        this.stackStatus = stackStatus;
    }

    public void setStackStatus(StackStatus stackStatus) {
        this.stackStatus = stackStatus.toString();
    }


Since this class comes from a third-party library, right now my only recourse is to create a proxy factory that produces objects with only one setter for any given property, which is pretty tedious.

**Comment by cowtowncoder** [comment]
Yes,this is unfortunate. If it was easy to fix, it would have, but I hope a solution will be eventually found.

**Comment by jharen** [comment]
Oh, for sure. Not a complaint, just logging another data point. I appreciate all the work you guys put in.

**Comment by cowtowncoder** [comment]
Looks like this actually works with 2.12, related to other changes I think. At least I can not reproduce the issue with simple changes here.

Also of note: #1044 added a way to resolve issues as well (included in 2.7).
